src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -2030,7 +2043,7 @@ type LspClient() =
     default __.TextDocumentPublishDiagnostics(_) = ignoreNotification
 
 [<AbstractClass>]
-type LspServer() =
+type LspServer(cancelRequest: (int -> unit)) =
```

Comment:
open question here: is there a better way to provide this delegate to the LspServer type? we need a way to do something in `x.CancelRequest`, since the request mapping is state internal to the `Server.start` function...

Reply:
there was a better way